### PR TITLE
[OpRefactor] Update kraus representations

### PR DESCRIPTION
**Pull request by mariaschuld**
**Context:**

The usual changes, this time for the kraus matrices

**Description of the Change:**

Renamed methods and changed kind of methods. Add docstrings everywhere.

### CODE REVIEW

File: doc/releases/changelog-dev.md
Comment (mariaschuld) [inline doc/releases/changelog-dev.md:250]:
```suggestion
  [(#2055)](https://github.com/PennyLaneAI/pennylane/pull/2055)
```

File: pennylane/operation.py
Comment (josh146) [inline pennylane/operation.py:1166]: I know we have already talked about this before, but is there any utility in moving this to `Operator`? Since I suppose anything can be represented using Kraus matrices right?

File: pennylane/ops/channel.py
Comment (josh146) [inline pennylane/ops/channel.py:67]:
```suggestion

        Returns:
```

File: pennylane/ops/channel.py
Comment (josh146) [inline pennylane/ops/channel.py:142]:
```suggestion

        Returns:
```

File: pennylane/ops/channel.py
Comment (josh146) [inline pennylane/ops/channel.py:210]:
```suggestion

        Returns:
```

File: pennylane/ops/channel.py
Comment (josh146) [inline pennylane/ops/channel.py:285]:
```suggestion
            list (array): list of Kraus matrices
```

File: pennylane/ops/channel.py
Comment (josh146) [inline pennylane/ops/channel.py:211]:
```suggestion
            list (array): list of Kraus matrices
```

File: pennylane/ops/channel.py
Comment (josh146) [inline pennylane/ops/channel.py:143]:
```suggestion
            list (array): list of Kraus matrices
```

File: pennylane/ops/channel.py
Comment (josh146) [inline pennylane/ops/channel.py:68]:
```suggestion
            list (array): list of Kraus matrices
```

File: pennylane/operation.py
Comment (josh146) [inline pennylane/operation.py:1202]:
```suggestion
            list (array): list of Kraus matrices
```

File: pennylane/operation.py
Comment (josh146) [inline pennylane/operation.py:1187]:
```suggestion
            list (array): list of Kraus matrices
```

File: pennylane/ops/channel.py
Comment (josh146) [inline pennylane/ops/channel.py:279]:
```suggestion
        r"""Kraus matrices representing the depolarizing channel.
```
reads a bit better? Might also apply to other methods above and below

File: pennylane/ops/channel.py
Comment (josh146) [inline pennylane/ops/channel.py:350]:
```suggestion
            list (array): list of Kraus matrices
```

File: pennylane/ops/channel.py
Comment (josh146) [inline pennylane/ops/channel.py:431]:
```suggestion
            list (array): list of Kraus matrices
```

File: pennylane/ops/channel.py
Comment (josh146) [inline pennylane/ops/channel.py:429]: What did L mean here?

File: pennylane/ops/channel.py
Comment (josh146) [inline pennylane/ops/channel.py:545]: Looks like this is missing!

File: pennylane/ops/channel.py
Comment (josh146) [inline pennylane/ops/channel.py:540]:
```suggestion
            list (array): list of Kraus matrices
```

File: pennylane/ops/channel.py
Comment (josh146) [inline pennylane/ops/channel.py:611]:
```suggestion
            list (array): list of Kraus matrices
```

File: pennylane/ops/channel.py
Comment (josh146) [inline pennylane/ops/channel.py:685]:
```suggestion
            list (array): list of Kraus matrices
```

File: pennylane/ops/channel.py
Comment (josh146) [inline pennylane/ops/channel.py:682]: lol I got very confused here for a moment.

In cases like this, do you think we should just write it like?
```python
class SomeChannel(Channel):
    compute_kraus_matrices = staticmethod(lambda x: x)
```

File: pennylane/ops/channel.py
Comment (josh146) [inline pennylane/ops/channel.py:792]:
```suggestion
            list (array): list of Kraus matrices
```

File: tests/test_operation.py
Comment (josh146) [inline tests/test_operation.py:1329]: looks so much cleaner!

**Review by josh146 — APPROVED** [review]
Thanks @mariaschuld! Have left various suggestions, most of them documentation related. Happy for this to be merged once addressed (don't think a second review by me is needed, unless major changes occur)

File: pennylane/operation.py
Comment (mariaschuld) [reply pennylane/operation.py:1166]: Yes...if I had more time I'd do it :) Maybe we hold off for now!

File: pennylane/ops/channel.py
Comment (mariaschuld) [reply pennylane/ops/channel.py:429]: Lol, no idea, but I must have overlooked it before when removing these things

File: pennylane/ops/channel.py
Comment (mariaschuld) [reply pennylane/ops/channel.py:545]: Thanks! I am getting tired!!

Comment (mariaschuld) [inline pennylane/ops/channel.py:545]:
```suggestion
        [array([[0.70710678, 0.        ], [0.        , 0.70710678]]), 
         array([[0.        , 0.70710678], [0.70710678, 0.        ]])]
```

File: pennylane/ops/channel.py
Comment (mariaschuld) [reply pennylane/ops/channel.py:682]: I'd like to have the docstring though :)

**Pull request merged**
